cpp/src/barretenberg/honk/proof_system/prover.cpp

```diff
@@ -66,6 +64,13 @@ Prover<settings>::Prover(std::vector<barretenberg::polynomial>&& wire_polys,
     prover_polynomials[POLYNOMIAL::W_L] = wire_polynomials[0];
     prover_polynomials[POLYNOMIAL::W_R] = wire_polynomials[1];
     prover_polynomials[POLYNOMIAL::W_O] = wire_polynomials[2];
+
+    // Add public inputs to transcript from the second wire polynomial
```

Comment:
Err, why is this public inputs stuff happening here now?

Reply:
This was something Adrian did that I think makes a lot of sense. First of all, the public inputs are needed in two places (add to transcript in some early round, then use to compute PI delta in grand product computation) so it makes sense to have a member variable `public_inputs`. Also, they're stored in the wires and we initialize the prover owned wires in this `Prover` constructor so I think it's natural to do the same for the PI. The final change (which was not necessary but makes good sense to me) was to add the PI to the transcript in the same round that we add `public_input_size`, rather than (seemingly arbitrarily) in `execute_wire_commitments_round`. I'm assuming there was originally some reason for this which has since gone away. Or maybe I'm missing a reason that still exists? Either way let's discuss

Reply:
Yeah, sounds reasonable, I just wasn't clear on why it's happening in this PR, if it was necessary I wanted to understand why.